### PR TITLE
Add SNAP inputs and update openfisca-US

### PR DESCRIPTION
**Pull request by MaxGhenis**
Adds more inputs to the US screen to improve SNAP calculation and update to the latest version of OpenFisca-US, which improves SNAP calculations. Specifically it includes:
* Earned income sources: employment & self-employment income
* Unearned income sources: dividend & interest income
* Medical expense deduction inputs: medical out of pocket expenses
* Inputs for special disability rules: SSDI, SSI disabled, and disabled veteran status or family member
* Categorical eligibility inputs: SSI
* Shelter deduction inputs: housing costs and homelessness
* Dependent care deduction inputs: childcare expenses (NB: this doesn't look good because we need units, see PolicyEngine/openfisca-us#496)

I omitted `spm_unit_assets` because we don't have stocks/flows yet in openfisca-us (to my knowledge?) and because California doesn't have an asset limit (due to BBCE).

Adding more parameters will require naming them in openfisca-us, plus in many cases a state drop-down.

I also removed UK-specific vanity URLs from the US page.

Here's how it looks. I had to create a SPM unit section, but we can fix that as @nikhilwoodruff is doing on the UK side to combine household-like entities.
image

### CODE REVIEW

File: policyengine-client/src/countries/uk/uk.jsx [hunk 1]
```diff
@@ -17,32 +17,32 @@ const childNamer = {
...
-                                    "https://policyengine.org/" :
-                                    `${window.location.protocol}//${window.location.hostname}`
-                            ) :
-                            `http://localhost:5000`;
+            (
+                this.usePolicyEngineOrgServer ?
+                    "https://policyengine.org/" :
+                    `${window.location.protocol}//${window.location.hostname}`
+            ) :
+            `http://127.0.0.1:5000`;
```
Comment (MaxGhenis) [inline policyengine-client/src/countries/uk/uk.jsx:45]: Does this address work on other machines too, or should I change it back to localhost?

**Comment by MaxGhenis** [comment]
I added a few more deductions named in openfisca-us 0.14.1, so this now depends on merging PolicyEngine/openfisca-us#498.

**Comment by MaxGhenis** [comment]
I tested this with the new deductions, looks good and produced expected results for a sample household!
image

File: policyengine-client/src/countries/uk/uk.jsx [hunk 2]
```diff
@@ -17,32 +17,32 @@ const childNamer = {
 }
 
 function validatePolicy(policy, defaultPolicy) {
-    if(defaultPolicy) {
-        for(let parameter in policy) {
+    if (defaultPolicy) {
```
Comment (MaxGhenis) [inline policyengine-client/src/countries/uk/uk.jsx:20]: Do you use a jsx formatter? This reformatting happened via my vscode's default.

Comment (nikhilwoodruff) [reply policyengine-client/src/countries/uk/uk.jsx:20]: No, which VS Code one are you using? I'll use that

File: policyengine-client/src/countries/uk/uk.jsx (hunk 1 above)
Comment (nikhilwoodruff) [reply policyengine-client/src/countries/uk/uk.jsx:45]: Just tested - works for me, I think it's OK to change it

**Review by nikhilwoodruff — APPROVED** [review]

**Pull request merged**